### PR TITLE
qbittorrent: build with qt6

### PR DESCRIPTION
**Pull request by slymattz**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - x86_64-musl (with -A flag)
  - aarch64 (crossbuild)

### CODE REVIEW

File: srcpkgs/qbittorrent/template [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-build_style=gnu-configure
-build_helper=qmake
-configure_args="--disable-systemd --with-boost=${XBPS_CROSS_BASE}/usr
- --with-boost-libdir=${XBPS_CROSS_BASE}/usr/lib"
-hostmakedepends="automake libtool pkg-config qt5-host-tools qt5-qmake qt5-tools"
-makedepends="libtorrent-rasterbar-devel qt5-declarative-devel qt5-svg-devel"
-depends="qt5-svg"
+build_style=cmake
+configure_args="-DQT6=ON -DSYSTEMD=OFF -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr -DGUI=ON -DWEBUI=OFF"
+hostmakedepends="cmake ninja pkg-config qt6-tools qt6-declarative-host-tools"
```
Comment (classabbyamp) [inline srcpkgs/qbittorrent/template:8]:
```suggestion
configure_args="-DQT6=ON -DSYSTEMD=OFF -DGUI=ON -DWEBUI=OFF"
hostmakedepends="pkg-config qt6-tools qt6-declarative-host-tools"
```
set by build style

File: srcpkgs/qbittorrent/template [hunk 2]
```diff
@@ -19,38 +17,35 @@ distfiles="${SOURCEFORGE_SITE}/qbittorrent/qbittorrent-${version}.tar.xz"
...
 CXXFLAGS=-std=gnu++17
 
-do_extract() {
-	local n=${pkgname}-${version}
-	mkdir -p build-nox build-x11
-	vsrcextract -C build-x11 ${n}.tar.xz
-	vsrcextract -C build-nox ${n}.tar.xz
+post_extract() {
+	mv ${pkgname}-${version}/* .
 }
```
Comment (classabbyamp) [inline srcpkgs/qbittorrent/template:22]: remove this function and create_wrksrc

File: srcpkgs/kid3/template
Comment (classabbyamp) [inline srcpkgs/kid3/template:1]: why?

Comment (slymattz) [reply srcpkgs/kid3/template:1]: oh gosh. sorry, this is purely a mess-up. I'll fix that.

File: srcpkgs/qbittorrent/template [hunk 3]
```diff
@@ -19,38 +16,31 @@ distfiles="${SOURCEFORGE_SITE}/qbittorrent/qbittorrent-${version}.tar.xz"
...
 CXXFLAGS=-std=gnu++17
 
-do_extract() {
-	local n=${pkgname}-${version}
-	mkdir -p build-nox build-x11
-	vsrcextract -C build-x11 ${n}.tar.xz
-	vsrcextract -C build-nox ${n}.tar.xz
+post_configure() { # qbittorrent-nox
+	configure_args=${configure_args//DGUI=ON/DGUI=OFF}
+	configure_args=${configure_args//DWEBUI=OFF/DWEBUI=ON}
```
Comment (classabbyamp) [inline srcpkgs/qbittorrent/template:21]:
```suggestion
	configure_args="${configure_args//DGUI=ON/DGUI=OFF}"
	configure_args="${configure_args//DWEBUI=OFF/DWEBUI=ON}"
```

File: srcpkgs/qbittorrent/template [hunk 4]
```diff
@@ -19,38 +16,31 @@ distfiles="${SOURCEFORGE_SITE}/qbittorrent/qbittorrent-${version}.tar.xz"
...
-do_configure() {
-	cd ${wrksrc}/build-x11
-	./bootstrap.sh
-	QT_QMAKE=${XBPS_WRAPPERDIR} ./configure ${configure_args} --enable-gui --disable-webui
-	cd ${wrksrc}/build-nox
-	./bootstrap.sh
-	QT_QMAKE=${XBPS_WRAPPERDIR} ./configure ${configure_args} --disable-gui --enable-webui
+post_build() { # qbittorrent-nox
+	(
+		DESTDIR=${DESTDIR/qbittorrent/qbittorrent-nox}
```
Comment (classabbyamp) [inline srcpkgs/qbittorrent/template:30]:
```suggestion
		DESTDIR="${DESTDIR/qbittorrent/qbittorrent-nox}"
```

File: srcpkgs/qbittorrent/template [hunk 5]
```diff
@@ -19,38 +16,31 @@ distfiles="${SOURCEFORGE_SITE}/qbittorrent/qbittorrent-${version}.tar.xz"
...
-	cd ${wrksrc}/build-nox
-	make ${makejobs} CC="$CC" CXX="$CXX" LINK="$CXX"
-}
-
-do_install() {
-	cd ${wrksrc}/build-x11
-	make ${makejobs} INSTALL_ROOT=${DESTDIR} install
+post_install() {
+	(
+		DESTDIR=${DESTDIR/qbittorrent/qbittorrent-nox}
```
Comment (classabbyamp) [inline srcpkgs/qbittorrent/template:38]:
```suggestion
		DESTDIR="${DESTDIR/qbittorrent/qbittorrent-nox}"
```

**Comment by slymattz** [comment]
I believe repo was not entirely synced because of 

> ERROR: [trans] failed to download `libglapi-24.1.4_1' signature from `https://repo-ci.voidlinux.org/current': Not Found

But it seems to be in sync now (x86_64-musl and armv6l-musl built fine). So can you re-run the workflow thingy please?

File: srcpkgs/qbittorrent/template [hunk 6]
```diff
@@ -19,38 +16,31 @@ distfiles="${SOURCEFORGE_SITE}/qbittorrent/qbittorrent-${version}.tar.xz"
...
-do_configure() {
-	cd ${wrksrc}/build-x11
-	./bootstrap.sh
-	QT_QMAKE=${XBPS_WRAPPERDIR} ./configure ${configure_args} --enable-gui --disable-webui
-	cd ${wrksrc}/build-nox
-	./bootstrap.sh
-	QT_QMAKE=${XBPS_WRAPPERDIR} ./configure ${configure_args} --disable-gui --enable-webui
+post_build() { # qbittorrent-nox
+	(
+		DESTDIR="${DESTDIR/qbittorrent/qbittorrent-nox}"
```
Comment (classabbyamp) [inline srcpkgs/qbittorrent/template:30]: don't think DESTDIR needs to be set for build

Comment (slymattz) [reply srcpkgs/qbittorrent/template]: done

**Pull request closed without merging**